util/io/files/tar.go

```diff
@@ -193,6 +198,9 @@ func (t *tgz) tgzFile(path string, fi os.FileInfo, err error) error {
 	}
 
 	header, err := tar.FileInfoHeader(fi, link)
+	// Write in PAX format to support sub-second time resolution.
+	// https://pkg.go.dev/archive/tar#Format
+	header.Format = tar.FormatPAX
```

Comment:
Is this absolutely necessary for your use case? I'm in favor of the smallest possible change to solve the problem.